utils/messages.ts

```diff
@@ -33,6 +41,37 @@ const trackErrorAndContinue = (
   return error;
 };
 
+const messageCategoryMapping = [
```

Comment:
What about using a record (ie. `{ category: MessageCategory, content: t.Type<Partial<MessageContent>> }`) instead of a Tuple? Consider that the Tuple types was included before Typescript tuples evolved, you probably have better choices now. Moreover Tuple2 it's not a monadic structure (you don't have `map()`, `reduce()`, ...)

You probably can use plain js records or tuples instead most of the time
(ie. `type MessageType = [MessageCategory, t.Type<Partial<MessageContent>>]`), in this way you can use fp-ts (Readonly)`Record` or (ReadOnly)`Tuple` utility methods.

I usually prefer records since it's easier to understand `mystruct.myField` than `mystruct[0]` or `mystruct.el1`